### PR TITLE
build: fixes

### PR DESCRIPTION
**Pull request by alichraghi**
- [x] By selecting this checkbox, I agree to license my contributions to this project under the license(s) described in the LICENSE file, and I have the right to do so or have received permission to do so by an employer or client I am producing work for whom has this right.

### CODE REVIEW

**Comment by alichraghi** [comment]
Closes #490

File: libs/ecs/build.zig [hunk 1]
```diff
@@ -9,6 +9,10 @@ pub fn build(b: *std.build.Builder) void {
 
 pub fn testStep(b: *std.build.Builder, mode: std.builtin.Mode, target: std.zig.CrossTarget) *std.build.RunStep {
     const main_tests = b.addTestExe("ecs-tests", (comptime thisDir()) ++ "/src/main.zig");
+
+    // TODO: remove this when tests passed with self-hosted compiler
```
Comment (slimsag) [inline libs/ecs/build.zig:13]:
```suggestion
    // TODO(self-hosted): remove this when tests passed with self-hosted compiler
```

**Review by slimsag — APPROVED** [review]

**Comment by mitchellh** [comment]
My only recommendation on this is out of an abundance of caution, I'd include at LEAST a `@sizeOf` test in these structs to ensure they're never broken. Its easy to do with padding miscalculations.

This will get better in stage2 with explicit integer-backed packed structs...

**Comment by alichraghi** [comment]
Don't worry. I did tested all structs carefully (not just size).
Also in case of invalid size, library won't compile. Because all flags are passed to functions, that expects a valid size.

**Pull request merged**